providers/default.rb

```diff
@@ -180,11 +184,15 @@ def run_post_install_action
 end
 
 def chef_install_dir
-  windows? ? 'c:/opscode/chef' : '/opt/chef'
+  if node['chef_client'].attribute?('bin')
```

Comment:
you should not go poking around the chef-client cookbook's node attributes here.  it creates tight coupling between those cookbooks.  it might seem convenient but the potential future confusion is not worth it (and i can visually see that 'chef-client' vs. 'chef_client' might cause issues in the future).